### PR TITLE
feat(xdg spec): use $XDG_DATA_HOME by default

### LINKED ISSUE #37: XDG_DIR spec
hello

the plugin create a ~/.supermaven folder, wish it could default to something like $XDG_DATA_HOME if it's defined, or at least have a option to change the folder

Thanks !

### PR DESCRIPTION
**Pull request by mirsella**
Seems the sm-agent binary still create the ~/.supermaven/config.json, sm-agent will need to be updated upstream

close #37

### CODE REVIEW

**Comment by flexagoon** [comment]
@sm-victorw can you please merge this? Would also be great if `sm-agent` created the config in the proper location (`~/.config/supermaven/config.json`) instead of `~/.supermaven`.

File: lua/supermaven-nvim/binary/binary_fetcher.lua [hunk 1]
```diff
@@ -112,7 +112,14 @@ end
 
 function BinaryFetcher:local_binary_parent_path()
   local home_dir = self.homedir
```
Comment (sm-victorw) [inline lua/supermaven-nvim/binary/binary_fetcher.lua:114]: Won't this value always be `nil`?

Comment (mirsella) [reply lua/supermaven-nvim/binary/binary_fetcher.lua:114]: i undo my change to the initialization of self.homedir

**Comment by flexagoon** [comment]
@sm-victorw since it's not open source and there's no way to contribute, can you tell if there are any plans to change sm-agent to use XDG location as well?

**Pull request merged**